### PR TITLE
feature #6646: support field()

### PR DESCRIPTION
**Pull request by YANGGMM**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6646 

## What this PR does / why we need it:
feature #6646: support field()

### CODE REVIEW

File: pkg/sql/plan/function/builtin/multi/field.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+		return nil, err
+	}
+	rs := vector.MustTCols[uint64](resultVector)
+
+	//if first vector is scalar
+	if firstVector.IsScalar() {
+
+		//if first vector is null, the return value is 0
+		if firstVector.IsScalarNull() {
+			return resultVector, nil
```
Comment (ouyuanning) [inline pkg/sql/plan/function/builtin/multi/field.go:48]: return vector.NewConstFixed is better?

File: pkg/sql/plan/function/builtin/multi/field.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
```
Comment (ouyuanning) [inline pkg/sql/plan/function/builtin/multi/field.go:163]: return scalar?

File: pkg/sql/plan/function/builtin/multi/field.go (hunk 1 above)
Comment (YANGGMM) [reply pkg/sql/plan/function/builtin/multi/field.go:48]: ok

File: pkg/sql/plan/function/builtin/multi/field.go (hunk 2 above)
Comment (YANGGMM) [reply pkg/sql/plan/function/builtin/multi/field.go:163]: ok

**Pull request closed without merging**